test/Microsoft.DotNet.Cli.Utils.Tests/GivenAProjectDependenciesCommandFactory.cs

```diff
@@ -147,5 +147,36 @@ public void It_resolves_desktop_apps_using_configuration_passed_to_create()
             command.CommandName.Should().Contain(Path.Combine(testInstance.TestRoot, "bin", configuration));
             Path.GetFileName(command.CommandName).Should().Be("dotnet-desktop-and-portable.exe");
         }
+
+        [WindowsOnlyFact]
+        public void It_resolves_tools_whose_package_name_is_different_than_dll_name()
+        {
+            var configuration = "Debug";
+
+            var testAssetManager = new TestAssetsManager(Path.Combine(RepoRoot, "TestAssets", "TestProjects"));
+            var testInstance = testAssetManager.CreateTestInstance("AppWithDirectDependencyWithOutputName")
+                .WithLockFiles();
+
+            var buildCommand = new BuildCommand(
+                Path.Combine(testInstance.TestRoot, "project.json"),
+                configuration: configuration)
+                    .ExecuteWithCapturedOutput()
+                    .Should()
+                    .Pass();
+
+            var context = ProjectContext.Create(testInstance.TestRoot, FrameworkConstants.CommonFrameworks.NetCoreApp10);
+
+            var factory = new ProjectDependenciesCommandFactory(
+                FrameworkConstants.CommonFrameworks.NetCoreApp10,
+                configuration,
+                null,
+                null,
+                testInstance.TestRoot);
+
+            var command = factory.Create("dotnet-tool-with-output-name", null);
+
+            command.CommandArgs.Should().Contain(
```

Comment:
@piotrpMSFT - should I be able to use `dotnet-dependency-tool-invoker` on non-windows? Because when running on Mac/Linux I see this: 
`Unable to resolve 'dotnet-dependency-tool-invoker (>= 1.0.0)'` and I also see this: https://github.com/dotnet/cli/blob/rel/1.0.0/test/dotnet.Tests/PackagedCommandTests.cs#L52-L55.
Seems to be working fine on Windows...